modules/astrodon-cli/commands/run.ts

```diff
@@ -1,66 +1,173 @@
+const DEFAULT_CONFIG: IAppConfig = {
+  name: "Astrodon",
+  id: "Astrodon",
+  main: "",
+  copyright: "2022",
+  version: "0.0.0",
+  author: "",
+  shortDescription: "",
+  longDescription: "",
+  homepage: "",
+  permissions: {
+    allow_hrtime: false,
+    prompt: false,
+  },
+  unstable: false,
+  build: {
+    output: "",
+    icons: [],
+    resources: [],
+  },
+};
+
 const runLogger = new Logger("run");
 
-export async function run(options?: RunOptions, file?: string) {
-  let projectInfo: IAppConfig = {
-    main: "",
-    name: "Astrodon",
-    id: "Astrodon",
-    copyright: new Date().getFullYear().toString(),
-    version: "0.0.0",
-    author: "",
-    shortDescription: "",
-    longDescription: "",
-    homepage: "",
-    permissions: {
-      allow_hrtime: true,
-      prompt: false,
-    },
-    unstable: false,
-    build: {
-      icons: [],
-      output: "",
-      resources: [],
-    },
-  };
-
-  if (file) {
-    projectInfo.main = file;
-  } else if (options) {
-    // Assume it's a local file by default
-    let filePath = `file://${resolve(Deno.cwd(), options.config)}`;
-
-    if (options.config.startsWith("http")) {
-      filePath = options.config;
-    }
+async function resolveConfiguration(
+  options: RunOptions,
+  file?: string,
+): Promise<IAppConfig | null> {
+
+  // Default Local path config
+  let configFile = `file://${
+    resolve(Deno.cwd(), dirname(file || ""), "astrodon.config.ts")
+  }`;
+
+  // Check if the user has provided a config file which is a local path
+  // And if it isn't an absolute path we will resolve it
+  if (file && !file?.startsWith("http")) {
+    file = isAbsolute(file) ? file : resolve(Deno.cwd(), file);
+  }
+
+  if (options.config?.startsWith("http")) {
+    // Custom HTTP path config
+    configFile = options.config;
+  } else if (file?.startsWith("http")) {
+    // Default HTTP path config
+    const fileUrl = new URL(file);
+    const { origin, pathname } = fileUrl;
+    const remoteDirname = dirname(pathname);
+    configFile = `${origin}${remoteDirname}/astrodon.config.ts`;
+  } else if (options.config) {
+    // Custom Local path config
+    configFile = `file://${
+      resolve(Deno.cwd(), dirname(file || ""), options.config)
+    }`;
+  }
+
+  const { href: configPath } = new URL(configFile);
+
+  try {
+    // Fetch the configuration file
+    const { default: projectInfo }: { default: IAppConfig } = await import(
+      configPath
+    );
 
-    const configPath = new URL(filePath).href;
+  
+    if (file) {
+      // Use the custom file if specified
+      return {
+        ...projectInfo,
+        main: file,
+      };
+    } else if (options.config?.startsWith("http")) {
+      // Use the relative entry file of the HTTP path config
+      const fileUrl = new URL(options.config);
+      const { origin, pathname } = fileUrl;
+      const remoteDirname = dirname(pathname);
+      projectInfo.main = `${origin}${remoteDirname}/${projectInfo.main}`;
```

Comment:
@marc2332 maybe we should resolve the `projectInfo.main` with the `remoteDirname` instead of just appending it to the url, I think it could produce something like this: `https://x.y/z/.main.ts` don't you think?